src/network/networkd-json.c

```diff
@@ -1161,6 +1162,143 @@ static int dns_misc_build_json(Link *link, JsonVariant **ret) {
         return r;
 }
 
+static int dhcp_server_offered_leases_build_json(Link *link, JsonVariant **ret) {
+        JsonVariant **elements = NULL;
+        size_t n = 0;
+        int r;
+        DHCPLease *lease;
+
+        assert(link);
+        assert(ret);
+
+        if (!link->dhcp_server) {
+                *ret = NULL;
+                return 0;
+        }
+
+        HASHMAP_FOREACH(lease, link->dhcp_server->bound_leases_by_client_id) {
+                _cleanup_free_ char *id = NULL;
+                if (!GREEDY_REALLOC(elements, n + 1)) {
+                        r = -ENOMEM;
+                        goto finalize;
+                }
+
+                struct in_addr address = {.s_addr = lease->address};
+
+                r = sd_dhcp_client_id_to_string(lease->client_id.data, lease->client_id.length, &id);
+                if (r < 0)
+                        return r;
+
+                dual_timestamp expiration;
+                dual_timestamp_from_boottime(&expiration, lease->expiration);
+
+                r = json_build(elements + n, JSON_BUILD_OBJECT(
+                        JSON_BUILD_PAIR_STRING("ClientId", id),
+                        JSON_BUILD_PAIR_IN4_ADDR_NON_NULL("Address", &address),
+                        JSON_BUILD_PAIR_UNSIGNED_NON_ZERO("Expiration", expiration.realtime / USEC_PER_SEC)));
```

Comment:
Please export the timestamp as is, instead of converting to realtime and seconds.

Reply:
I guess it can be calculated in whatever reads the JSON by getting the CLOCK_BOOTTIME from the system.